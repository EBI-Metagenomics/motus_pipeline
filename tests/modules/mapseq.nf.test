nextflow_process {

    name "Test module mapseq"
    script "modules/mapseq.nf"
    process "MAPSEQ"

    test("mapseq") {

        when {
            params {
                seq = "${baseDir}/tests/modules/fixtures/mapseq/test.SSU.fasta"
                db = "${baseDir}/tests/modules/fixtures/mapseq/silva-SSU/silva-SSU.fasta"
                tax = "${baseDir}/tests/modules/fixtures/mapseq/silva-filt.txt"
            }
            process {
                """
                input[0] = channel.fromPath("${params.seq}", checkIfExists: true)
                input[1] = channel.fromPath("${params.db}", checkIfExists: true)
                input[2] = channel.fromPath("${params.tax}", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}