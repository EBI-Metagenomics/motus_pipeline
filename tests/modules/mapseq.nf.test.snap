{
    "mapseq": {
        "content": [
            {
                "mapseq_result": [
                    "test.SSU_silva-SSU.mseq:md5,ccc9d418d533cb4b0966ad46283bcd5b"
                ]
            }
        ],
        "timestamp": "2023-02-28T16:07:42+0000"
    }
}